models/classes/CrudResultsService.php

```diff
@@ -62,21 +59,13 @@ public function get($uri, $groupBy = self::GROUP_BY_DELIVERY)
         return $this->format($implementation, $uri);
     }
 
-    /**
-     * @param taoResultServer_models_classes_ReadableResultStorage $resultStorage
-     * @param $resultIdentifier
-     * @param int $groupBy
-     * @param bool $fetchOnlyLastAttemptResult
-     * @return array
-     * @throws common_exception_Error
-     */
     public function format(
         taoResultServer_models_classes_ReadableResultStorage $resultStorage,
         $resultIdentifier,
-        $groupBy = self::GROUP_BY_DELIVERY,
-        $fetchOnlyLastAttemptResult = false
-    )
-    {
+        int $groupBy = self::GROUP_BY_DELIVERY,
+        bool $fetchOnlyLastAttemptResult = false,
+        bool $splitByAttempt = false
```

Comment:
Instead of having this new bool value, you can use the $groupBy to do the decision internally.

- [ ] Consider removing unnecessary $splitByAttempt variable

Reply:
Do you mean bit masks to allow combinations of the grouping or just dedicated option for the ‘item with attempts’ breakdown ?

Reply:
I mean just avoid a new boolean parameter that tells which internal formatting step should be taken since this is already based on the $groupBy parameter.

As far as I see every time the `$groupBy` is `CrudResultsService::GROUP_BY_ITEM` we should split the attempt, no?

Reply:
I need to check if the existent logic is not used somewhere else to avoid unpredictable changes. As I’ve altered the structure of the returned array that may break some other classes